sklearn/model_selection/tests/test_split.py

```diff
@@ -96,6 +96,10 @@
 )
 digits = load_digits()
 
+pytestmark = pytest.mark.filterwarnings(
```

Comment:
It is a bit weird for me to have this filter globally set for the file.

We should have our test catching the warning to make sure that we raise it when we expect or otherwise, we should raise an error.

Is there a particular reason to go fine grain here?

Reply:
I converted this `pytestmark` to error instead and updated the tests.